<a href="https://colab.research.google.com/github/dyou-wbg/meta_processing/blob/main/Clean__metadata_TitleBasedRCTclassifier.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')
import pandas as pd
# as of today
import datetime as dt
dt.date.today()
import os
import pandas as pd
from google.colab import drive
import csv
import os


Mounted at /content/drive


In [ ]:
df.head()

filename  \
0  http-155-185-68-2-recentpaper-recent-wp127-pdf...   
1              http-155-185-68-2-wpdemb-0168-pdf.pdf   
2  http-193-0-126-150-portalefiles-3842-journal-o...   
3  http-50sea-com-wp-content-uploads-2019-06-ayes...   
4  http-50sea-com-wp-content-uploads-2019-07-e-go...   

                                               title      type  \
0  The Macroeconomics Outcome of Oil Shocks in th...  preprint   
1  Tertiary education decisions of immigrants and...  preprint   
2                      Deep Roots of Fiscal Behavior   article   
3  Estimation of Wheat Area using Sentinel-1 and ...   article   
4  Role of Inter-Organizational Information Integ...   article   

                                             journal     date  \
0                                                NaN  2017-02   
1                                                NaN  2020-03   
2         Journal of Banking and Financial Economics     2014   
3  International Journal of Agriculture and Susta...     2019   
4  International Journal of Innovations in Scienc...     2019   

                                                 url  
0   http://155.185.68.2/Recentpaper/recent-wp127.pdf  
1                http://155.185.68.2/wpdemb/0168.pdf  
2  http://193.0.126.150/portaleFiles/3842-journal...  
3  http://50sea.com/wp-content/uploads/2019/06/Ay...  
4  http://50sea.com/wp-content/uploads/2019/07/E-...

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/metadata_jstor.csv', nrows=10)
df2.head()

id  \
0                http://www.jstor.org/stable/1182553   
1                http://www.jstor.org/stable/1030535   
2  http://www.jstor.org/stable/10.5325/transporta...   
3         http://www.jstor.org/stable/10.1086/587909   
4                http://www.jstor.org/stable/1094350   

                                               title  \
0  Economic Education in Our Schools: A Renewed M...   
1                 Can Germany Contain Russia Safely?   
2  Assessing the Impact of Sarbanes-Oxley Act on ...   
3  Childhood Educational Disruption and Later Lif...   
4          The Exclusion of Liability for Negligence   

                                            isPartOf  publicationYear  doi  \
0                  The Journal of Economic Education             1984  NaN   
1  The Annals of the American Academy of Politica...             1952  NaN   
2                             Transportation Journal             2014  NaN   
3                           Journal of Human Capital             2008  NaN   
4                              The Modern Law Review             1974  NaN   

   docType        docSubType provider  collection datePublished  ...  \
0  article  research-article    jstor         NaN    1984-10-01  ...   
1  article  research-article    jstor         NaN    1952-07-01  ...   
2  article  research-article    jstor         NaN    2014-01-01  ...   
3  article  research-article    jstor         NaN    2008-06-01  ...   
4  article  research-article    jstor         NaN    1974-03-01  ...   

                                                 url  \
0                http://www.jstor.org/stable/1182553   
1                http://www.jstor.org/stable/1030535   
2  http://www.jstor.org/stable/10.5325/transporta...   
3         http://www.jstor.org/stable/10.1086/587909   
4                http://www.jstor.org/stable/1094350   

                               creator                        publisher  \
0                     Richard M. Cyert           Taylor & Francis, Ltd.   
1                     Joseph C. Harsch          Sage Publications, Inc.   
2  Mahesh Srinivasan; Akhilesh Chandra      Penn State University Press   
3                          Paul Heaton  The University of Chicago Press   
4                        J. R. M. Lowe                Modern Law Review   

  language pageStart pageEnd  placeOfPublication  wordCount  pageCount  \
0      eng       261     264                 NaN       1824          4   
1      eng        36      40                 NaN       2570          5   
2      eng        44      78                 NaN       9922         35   
3      eng       154     187                 NaN      11759         34   
4      eng       218     221                 NaN       2072          4   

              file  
0  part-1.jsonl.gz  
1  part-1.jsonl.gz  
2  part-1.jsonl.gz  
3  part-1.jsonl.gz  
4  part-1.jsonl.gz  

[5 rows x 22 columns]

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/metadata_scopus 2.csv', nrows=10)
df2.head()

Unnamed: 0.1  Unnamed: 0              scopus_ID  \
0             0           0  SCOPUS_ID:85101100172   
1             1           1  SCOPUS_ID:85096201068   
2             2           2  SCOPUS_ID:85056885088   
3             5           5  SCOPUS_ID:84992147324   
4             6           6  SCOPUS_ID:85089158368   

                                         description   coverDate  \
0  © 2021 by ASERS® Publishing. All rights rese...  2021-01-01   
1  © 2020 Elsevier LtdMaritime Knowledge Cluster...  2021-01-01   
2  © 2017, Copyright © Taylor & Francis Group, ...  2017-10-02   
3  © 2016 Elsevier B.V.Changes in residual volat...  2016-12-01   
4  © 2020 Elsevier B.V.This study documents an u...  2020-10-01   

  aggregationType                                                url  \
0         Journal  https://api.elsevier.com/content/abstract/scop...   
1         Journal  https://api.elsevier.com/content/abstract/scop...   
2         Journal  https://api.elsevier.com/content/abstract/scop...   
3         Journal  https://api.elsevier.com/content/abstract/scop...   
4         Journal  https://api.elsevier.com/content/abstract/scop...   

     source-id  citedby-count  \
0  21100317750              0   
1        27851              4   
2  21100886534              3   
3        28976             17   
4        17496              2   

                                               title  openaccess  \
0  The marketing efficiency development to create...         NaN   
1  Maritime knowledge clusters: A conceptual mode...         0.0   
2  From “Communautaire Spirit” to the “Ghos...         1.0   
3    Testing for identification in SVAR-GARCH models         2.0   
4  Move to success? Headquarters relocation, poli...         0.0   

  subtypeDescription                                  publicationName  \
0            Article  Journal of Environmental Management and Tourism   
1            Article                                    Marine Policy   
2            Article       International Journal of Political Economy   
3            Article         Journal of Economic Dynamics and Control   
4            Article                     Journal of Corporate Finance   

                              doi  abstracts language  
0      10.14505/jemt.v12.1(49).23        NaN      eng  
1    10.1016/j.marpol.2020.104299        NaN      eng  
2   10.1080/08911916.2017.1407737        NaN      eng  
3      10.1016/j.jedc.2016.09.007        NaN      eng  
4  10.1016/j.jcorpfin.2020.101698        NaN      eng

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/elsevier_metadata.csv', nrows=10)
df2.head()

Unnamed: 0  id     type                                              title  \
0           0   0  article  Academic versus industry patenting: An in-dept...   
1           1   1  article  Unpacking transnational industry legitimacy dy...   
2           2   2  article  Influencing scientists' collaboration and prod...   
3           3   3  article  New forms of organisation and R&D internationa...   
4           4   4  article  Prospective analysis of technological innovati...   

                                            abstract  \
0                                                NaN   
1  This paper posits that the combination of chan...   
2  This paper analyzes the effect of university r...   
3  This paper explores the strategic internationa...   
4  In this article, we put forward a concept for ...   

                                            keywords  \
0                                                NaN   
1  Catch-up; Complex product systems; Industry le...   
2  University research center Research collaborat...   
3  Food and beverages Multinational enterprises R...   
4  Variation Technological innovation systems Dev...   

                                             authors    date  \
0  Sapsalis, Eleftherios, van Pottelsberghe de la...  2006.0   
1                Kwak, Kiho, Yoon, Hyungseok (David)  2020.0   
2          Ponomariov, Branco L., Boardman, P. Craig  2010.0   
3  Filippaios, Fragkiskos, Papanastassiou, Marina...  2009.0   
4  Markard, Jochen, Stadelmann, Martin, Truffer, ...  2009.0   

                                                 url          journal  \
0  http://www.sciencedirect.com/science/article/p...  research policy   
1  http://www.sciencedirect.com/science/article/p...  research policy   
2  http://www.sciencedirect.com/science/article/p...  research policy   
3  http://www.sciencedirect.com/science/article/p...  research policy   
4  http://www.sciencedirect.com/science/article/p...  research policy   

   downloaded  
0        True  
1        True  
2        True  
3        True  
4        True

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/url1_ie_table_complete.csv', nrows=10)
df2.head()

Unnamed: 0   id                                              title  \
0         0.0  0.0  Impacts of payments for ecosystem services pro...   
1         1.0  1.0  Providing Collateral And Improving Product Mar...   
2         2.0  2.0  Seeing Is Believing? Evidence From An Extensio...   
3         3.0  3.0  Smallholder Access To Weather Securities In In...   
4         4.0  4.0  Sustainability Of Impact Dimensions Of Decline...   

                                              author  \
0  Alix-Garcia, Jennifer; Aronson, Glen; Radeloff...   
1  Casaburi, Lorenzo; Glennerster, Rachel; Suri, ...   
2  Kondylis, Florence; Mueller, Valerie; Zhu, Siyao;   
3  Ceballos, Francisco; R, Isaac Manuel; Robles, ...   
4  Mcniven, Scott; Gilligan, Daniel O; Hotz, Chri...   

                               sector       country  \
0  Agriculture, fishing, and forestry        Mexico   
1  Agriculture, fishing, and forestry  Sierra Leone   
2  Agriculture, fishing, and forestry    Mozambique   
3  Agriculture, fishing, and forestry         India   
4  Agriculture, fishing, and forestry        Uganda   

                                                 URL  pub_time  \
0  https://developmentevidence.3ieimpact.org/sear...      2015   
1  https://developmentevidence.3ieimpact.org/sear...      2014   
2  https://developmentevidence.3ieimpact.org/sear...      2014   
3  https://developmentevidence.3ieimpact.org/sear...      2015   
4  https://developmentevidence.3ieimpact.org/sear...      2016   

                                                 pub  open_access  \
0  ['Publication Details', '3ie Series Report', '...         True   
1  ['Publication Details', '3ie Series Report', '...         True   
2  ['Publication Details', '3ie Grantee Final Rep...         True   
3  ['Publication Details', '3ie Series Report', '...         True   
4  ['Publication Details', '3ie Series Report', '...         True   

                                            category  \
0  ['Intervention: Forest conservation policy', '...   
1  ['Intervention 1: Agricultural market informat...   
2  ['Intervention: Introduction of agricultural e...   
3  ['Intervention: Index insurance + Introduction...   
4  ['Intervention: Crop management training + See...   

                                            abstract  \
0                                        No abstract   
1  The report presents the results from the evalu...   
2  Extension is designed to enable lab-to-farm te...   
3  Weather-based insurance products insure farmer...   
4  We examine the sustainability of the impact of...   

                                              panels  \
0  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
1  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
2  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
3  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
4  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   

                                          source_url  
0  http://3ieimpact.org/evidence-hub/publications...  
1                   https://doi.org/10.23846/ow31180  
2  https://www.3ieimpact.org/sites/default/files/...  
3  https://www.3ieimpact.org/evidence-hub/publica...  
4  http://3ieimpact.org/evidence-hub/publications...

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/metadata_econlit.csv', nrows=10)
df2.head()

recordID                                            pdfLink  \
0       141  https://content.ebscohost.com/cds/retrieve?con...   
1       142  https://content.ebscohost.com/cds/retrieve?con...   
2       143  https://content.ebscohost.com/cds/retrieve?con...   
3       144  https://content.ebscohost.com/cds/retrieve?con...   
4       145  https://content.ebscohost.com/cds/retrieve?con...   

                                               plink shortDbName  \
0  https://search.ebscohost.com/login.aspx?direct...         eoh   
1  https://search.ebscohost.com/login.aspx?direct...         eoh   
2  https://search.ebscohost.com/login.aspx?direct...         eoh   
3  https://search.ebscohost.com/login.aspx?direct...         eoh   
4  https://search.ebscohost.com/login.aspx?direct...         eoh   

                                           journal  pub_date  \
0   Capitalism: A Journal of History and Economics  20221201   
1  Central European Journal of Operations Research  20221201   
2  Central European Journal of Operations Research  20221201   
3  Central European Journal of Operations Research  20221201   
4  Central European Journal of Operations Research  20221201   

                                                 doi  \
0                                            1969735   
1  ['2000323', {'@type': 'doi', '#text': '10.1007...   
2  ['2000325', OrderedDict([('@type', 'doi'), ('#...   
3  ['2000326', OrderedDict([('@type', 'doi'), ('#...   
4  ['2000329', {'@type': 'doi', '#text': '10.1007...   

                                               title  \
0  Microcosms and Macrocosms: Capitalism and the ...   
1     Stochastic Radiotherapy Appointment Scheduling   
2  Examining the Balance between Efficiency and R...   
3           The Variable-Width Strip Packing Problem   
4  Constrained Naive Bayes with Application to Un...   

                                              author  \
0                                     Biltoft, C. N.   
1  ['Braune, Roland', 'Gutjahr, Walter J.', 'Vogl...   
2  ['de Arquer, Maria', 'Ponte, Borja', 'Pino, Ra...   
3                 ['Bodis, Attila', 'Csirik, Janos']   
4  ['Blanquero, Rafael', 'Carrizosa, Emilio', 'Ra...   

                                            abstract  Unnamed: 0.1  
0  Using the metaphor of microcosm and macrocosm,...           NaN  
1  When scheduling the starting times for treatme...           NaN  
2  We study a hybrid system where the demand of c...           NaN  
3  Here, we focus on a generalized version of the...           NaN  
4  The Naive Bayes is a tractable and efficient a...           NaN

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/SSRN_metadata.csv', nrows=10)
df2.head()

Unnamed: 0  id                                           doc_name  \
0           0   0  The Conclusion and Termination of the 'First' ...   
1           1   1  Investing in School Readiness: An Analysis of ...   
2           2   2  Science, Specific Knowledge and Total Quality ...   
3           3   3                       To Believe or Not to Believe   
4           5   5  Self Interest, Altruism, Incentives, and Agenc...   

   posted_date                                            authors  \
0  29 Sep 0012                                   Sunita Jogarajan   
1  09 Jan 0017  Nozomi Nakajima, Amer Hasan, Haeil  Jung, Sall...   
2    11-Jul-94                  Karen H. Wruck, Michael C. Jensen   
3    25-Jul-94   Utpal Bhattacharya , Murugappa (Murgie) Krishnan   
4     5-Oct-94                                  Michael C. Jensen   

                                             doc_url  
0  https://papers.ssrn.com/sol3/papers.cfm?abstra...  
1  https://papers.ssrn.com/sol3/papers.cfm?abstra...  
2  https://papers.ssrn.com/sol3/papers.cfm?abstra...  
3  https://papers.ssrn.com/sol3/papers.cfm?abstra...  
4  https://papers.ssrn.com/sol3/papers.cfm?abstra...

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/log.csv', nrows=10)
df2.head()

iteration working_paper_id pdf_name  \
0          1            w0001    w0001   
1          2            w0002    w0002   
2          3            w0003    w0003   
3          4            w0004    w0004   
4          5            w0005    w0005   

                                             pdf_url  \
0  https://www.nber.org/system/files/working_pape...   
1  https://www.nber.org/system/files/working_pape...   
2  https://www.nber.org/system/files/working_pape...   
3  https://www.nber.org/system/files/working_pape...   
4  https://www.nber.org/system/files/working_pape...   

                 retrieved_at                   result  
0  2022-05-27 14:57:01.807231  successfully retrieved!  
1  2022-05-27 14:57:03.603092  successfully retrieved!  
2  2022-05-27 14:57:05.103170  successfully retrieved!  
3  2022-05-27 14:57:06.977062  successfully retrieved!  
4  2022-05-27 14:57:08.292688  successfully retrieved!

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/metadata_semantic_scholar.csv', nrows=10)
df2.head()

paper_id  \
0  0000003ee9b8a4ef644842693ba78ce420cdc439   
1  000001d91985c527a5964dda397210f0eae2ecfd   
2  0000047b1466febc0389fc681456ee858e73d179   
3  000005775773f34cd0e89135b3221ffd5a34eea2   
4  000005f7ebe1a860b19ed9b82fd42361be05b7fe   

                                               title  \
0                Agro-Industries for Economic Growth   
1  Sequential Auctions with Supply or Demand Unce...   
2  Comparative performance of IRDP schemes in gen...   
3  Illiquidity in Intermediary Portfolios: Eviden...   
4  Moving Back Home: Insurance Against Labor Mark...   

                                               venue    year  reference_count  \
0  SEDME (Small Enterprises Development, Manageme...  1980.0                0   
1                                                NaN  1997.0                0   
2                                                NaN  1990.0                0   
3                                                NaN  2018.0                0   
4                                                NaN  2008.0               60   

   citation_count publication_date publication_types  \
0               0       1980-06-01    JournalArticle   
1              12              NaN               NaN   
2               0              NaN               NaN   
3               9       2018-01-05               NaN   
4               0              NaN               NaN   

                                     authors   s2_fields_of_study  \
0                                   N. Joshi            Economics   
1                    R. B. Verde;J. Sákovics  Economics;Economics   
2  D. A. Deshmukh;S. Kalyankar;K. D. Rajmane  Economics;Economics   
3                  D. Barth;Phillip J. Monin   Business;Economics   
4                                Greg Kaplan            Economics   

  fields_of_study                                       journal_name  \
0             NaN  SEDME (Small Enterprises Development, Manageme...   
1       Economics                                                NaN   
2       Economics                                                NaN   
3        Business                    Social Science Research Network   
4             NaN                                                NaN   

         mag_id  corpus_id  pub_med_id  pub_med_central_id  arxiv_id  dblp_id  \
0           NaN        NaN         NaN                 NaN       NaN      NaN   
1  2.793010e+09        NaN         NaN                 NaN       NaN      NaN   
2  3.169189e+08        NaN         NaN                 NaN       NaN      NaN   
3  2.763354e+09        NaN         NaN                 NaN       NaN      NaN   
4           NaN        NaN         NaN                 NaN       NaN      NaN   

   acl_id                       doi  
0     NaN  10.1177/0970846419800204  
1     NaN                       NaN  
2     NaN                       NaN  
3     NaN      10.2139/SSRN.3044468  
4     NaN                       NaN

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/url1_ie_table_complete.csv', nrows=10)
df2.head()


Unnamed: 0   id                                              title  \
0         0.0  0.0  Impacts of payments for ecosystem services pro...   
1         1.0  1.0  Providing Collateral And Improving Product Mar...   
2         2.0  2.0  Seeing Is Believing? Evidence From An Extensio...   
3         3.0  3.0  Smallholder Access To Weather Securities In In...   
4         4.0  4.0  Sustainability Of Impact Dimensions Of Decline...   

                                              author  \
0  Alix-Garcia, Jennifer; Aronson, Glen; Radeloff...   
1  Casaburi, Lorenzo; Glennerster, Rachel; Suri, ...   
2  Kondylis, Florence; Mueller, Valerie; Zhu, Siyao;   
3  Ceballos, Francisco; R, Isaac Manuel; Robles, ...   
4  Mcniven, Scott; Gilligan, Daniel O; Hotz, Chri...   

                               sector       country  \
0  Agriculture, fishing, and forestry        Mexico   
1  Agriculture, fishing, and forestry  Sierra Leone   
2  Agriculture, fishing, and forestry    Mozambique   
3  Agriculture, fishing, and forestry         India   
4  Agriculture, fishing, and forestry        Uganda   

                                                 URL  pub_time  \
0  https://developmentevidence.3ieimpact.org/sear...      2015   
1  https://developmentevidence.3ieimpact.org/sear...      2014   
2  https://developmentevidence.3ieimpact.org/sear...      2014   
3  https://developmentevidence.3ieimpact.org/sear...      2015   
4  https://developmentevidence.3ieimpact.org/sear...      2016   

                                                 pub  open_access  \
0  ['Publication Details', '3ie Series Report', '...         True   
1  ['Publication Details', '3ie Series Report', '...         True   
2  ['Publication Details', '3ie Grantee Final Rep...         True   
3  ['Publication Details', '3ie Series Report', '...         True   
4  ['Publication Details', '3ie Series Report', '...         True   

                                            category  \
0  ['Intervention: Forest conservation policy', '...   
1  ['Intervention 1: Agricultural market informat...   
2  ['Intervention: Introduction of agricultural e...   
3  ['Intervention: Index insurance + Introduction...   
4  ['Intervention: Crop management training + See...   

                                            abstract  \
0                                        No abstract   
1  The report presents the results from the evalu...   
2  Extension is designed to enable lab-to-farm te...   
3  Weather-based insurance products insure farmer...   
4  We examine the sustainability of the impact of...   

                                              panels  \
0  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
1  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
2  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
3  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
4  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   

                                          source_url  
0  http://3ieimpact.org/evidence-hub/publications...  
1                   https://doi.org/10.23846/ow31180  
2  https://www.3ieimpact.org/sites/default/files/...  
3  https://www.3ieimpact.org/evidence-hub/publica...  
4  http://3ieimpact.org/evidence-hub/publications...

In [ ]:


import pandas as pd
directory = '/content/drive/MyDrive/finalMetaFile'
final_file_list=['metadata_jstor.csv','metadata_scopus 2.csv','elsevier_metadata.csv','metadata_semantic_scholar.csv','url1_ie_table_complete.csv','metadata_econlit.csv','SSRN_metadata.csv','log.csv','metadata_repec.csv']

dataframes = []

for filename in final_file_list:
  if os.path.isfile(os.path.join(directory, filename)) and filename.endswith(".csv"):
    df = pd.read_csv(os.path.join(directory, filename))
    if "title" in df.columns:
      df['source']=filename
      df_RCT=df[df['title'].str.contains('random', case=False, regex=True, na=False)]
      dataframes.append(df_RCT[["title",'source']])
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))
    elif 'doc_name' in df.columns:
      df_RCT=df[df['doc_name'].str.contains('random', case=False, regex=True, na=False)]
      dataframes.append(df_RCT)
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))

df_combined = pd.concat(dataframes, ignore_index=True)
df_combined.to_csv('/content/drive/MyDrive/RCT_title_all_2.csv')


<ipython-input-3-8714c1c0bf44>:11: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_jstor.csv : 1095398
metadata_jstor.csv : 0.0012589031566608666
metadata_scopus 2.csv : 711784
metadata_scopus 2.csv : 0.0034195767255234735
elsevier_metadata.csv : 242410
elsevier_metadata.csv : 0.0040757394496926695


<ipython-input-3-8714c1c0bf44>:11: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_semantic_scholar.csv : 7912377
metadata_semantic_scholar.csv : 0.0015926945846993894
url1_ie_table_complete.csv : 11377
url1_ie_table_complete.csv : 0.23222290586270547
metadata_econlit.csv : 19121
metadata_econlit.csv : 0.005909732754563046
SSRN_metadata.csv : 99106
SSRN_metadata.csv : 0.0028454382176659335


<ipython-input-3-8714c1c0bf44>:11: DtypeWarning: Columns (4,7) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_repec.csv : 915771
metadata_repec.csv : 0.0038666871958164214


* metadata_jstor.csv : 0.12% (0.0012589031566608666)
* metadata_scopus 2.csv : 0.3% (0.0034195767255234735)
* elsevier_metadata.csv : 0.41% (0.0040757394496926695)
* metadata_semantic_scholar.csv : 0.15% (0.0015926945846993894)
* url1_ie_table_complete.csv : 23% (0.23222290586270547)
* metadata_econlit.csv : 0.59% (0.005909732754563046)
* SSRN_metadata.csv : 0.28% (0.0028454382176659335)
* metadata_repec.csv : 0.38% (0.0038666871958164214)

In [ ]:


import pandas as pd
directory = '/content/drive/MyDrive/finalMetaFile'
final_file_list=['metadata_jstor.csv','metadata_scopus 2.csv','elsevier_metadata.csv','metadata_semantic_scholar.csv','url1_ie_table_complete.csv','metadata_econlit.csv','SSRN_metadata.csv','log.csv','metadata_repec.csv']

dataframes = []

for filename in final_file_list:
  if os.path.isfile(os.path.join(directory, filename)) and filename.endswith(".csv"):
    df = pd.read_csv(os.path.join(directory, filename))
    if "title" in df.columns:
      df['source']=filename
      df_RCT=df[df['title'].str.contains('random|RCT', case=False, regex=True, na=False)]
      dataframes.append(df_RCT[["title",'source']])
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))
    elif 'doc_name' in df.columns:
      df_RCT=df[df['doc_name'].str.contains('random|RCT', case=False, regex=True, na=False)]
      dataframes.append(df_RCT)
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))

df_combined = pd.concat(dataframes, ignore_index=True)
df_combined.to_csv('/content/drive/MyDrive/RCT_title_all_2.csv')


<ipython-input-4-8fedeb9da70a>:11: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_jstor.csv : 1095398
metadata_jstor.csv : 0.001844078590612727
metadata_scopus 2.csv : 711784
metadata_scopus 2.csv : 0.003908489092196509
elsevier_metadata.csv : 242410
elsevier_metadata.csv : 0.0046408976527370985


<ipython-input-4-8fedeb9da70a>:11: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_semantic_scholar.csv : 7912377
metadata_semantic_scholar.csv : 0.0019137106333532894
url1_ie_table_complete.csv : 11377
url1_ie_table_complete.csv : 0.2345961149687967
metadata_econlit.csv : 19121
metadata_econlit.csv : 0.006014329794466817
SSRN_metadata.csv : 99106
SSRN_metadata.csv : 0.0030876031723609063


<ipython-input-4-8fedeb9da70a>:11: DtypeWarning: Columns (4,7) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_repec.csv : 915771
metadata_repec.csv : 0.004314397376636736



* metadata_jstor.csv : 1095398
* metadata_jstor.csv : 0.001844078590612727
* metadata_scopus 2.csv : 711784
* metadata_scopus 2.csv : 0.003908489092196509
* elsevier_metadata.csv : 242410
* elsevier_metadata.csv : 0.0046408976527370985
* metadata_semantic_scholar.csv : 7912377
* metadata_semantic_scholar.csv : 0.0019137106333532894
* url1_ie_table_complete.csv : 11377
* url1_ie_table_complete.csv : 0.2345961149687967
* metadata_econlit.csv : 19121
* metadata_econlit.csv : 0.006014329794466817
* SSRN_metadata.csv : 99106
* SSRN_metadata.csv : 0.0030876031723609063
* metadata_repec.csv : 915771
* metadata_repec.csv : 0.004314397376636736

In [ ]:


import pandas as pd
directory = '/content/drive/MyDrive/finalMetaFile'
final_file_list=['metadata_jstor.csv','metadata_scopus 2.csv','elsevier_metadata.csv','metadata_semantic_scholar.csv','url1_ie_table_complete.csv','metadata_econlit.csv','SSRN_metadata.csv','log.csv','metadata_repec.csv']

dataframes = []

for filename in final_file_list:
  if os.path.isfile(os.path.join(directory, filename)) and filename.endswith(".csv"):
    df = pd.read_csv(os.path.join(directory, filename))
    if "title" in df.columns:
      df['source']=filename
      df_RCT=df[df['title'].str.contains('random|RCT|randomized', case=False, regex=True, na=False)]
      dataframes.append(df_RCT[["title",'source']])
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))
    elif 'doc_name' in df.columns:
      df_RCT=df[df['doc_name'].str.contains('random|RCT|randomized', case=False, regex=True, na=False)]
      dataframes.append(df_RCT)
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))

df_combined = pd.concat(dataframes, ignore_index=True)
df_combined.to_csv('/content/drive/MyDrive/RCT_title_all_3keywords.csv')


<ipython-input-5-2edf79bb28da>:9: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_jstor.csv : 1095398
metadata_jstor.csv : 0.001844078590612727
metadata_scopus 2.csv : 711784
metadata_scopus 2.csv : 0.003908489092196509
elsevier_metadata.csv : 242410
elsevier_metadata.csv : 0.0046408976527370985


<ipython-input-5-2edf79bb28da>:9: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_semantic_scholar.csv : 7912377
metadata_semantic_scholar.csv : 0.0019137106333532894
url1_ie_table_complete.csv : 11377
url1_ie_table_complete.csv : 0.2345961149687967
metadata_econlit.csv : 19121
metadata_econlit.csv : 0.006014329794466817
SSRN_metadata.csv : 99106
SSRN_metadata.csv : 0.0030876031723609063


<ipython-input-5-2edf79bb28da>:9: DtypeWarning: Columns (4,7) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_repec.csv : 915771
metadata_repec.csv : 0.004314397376636736
